ksmt-core/src/main/kotlin/io/ksmt/expr/rewrite/simplify/ArithSimplification.kt

```diff
@@ -157,12 +154,47 @@ fun <T : KArithSort> KContext.simplifyArithDiv(lhs: KExpr<T>, rhs: KExpr<T>): KE
     return mkArithDivNoSimplify(lhs, rhs)
 }
 
-@Suppress("ForbiddenComment")
 fun <T : KArithSort> KContext.simplifyArithPower(lhs: KExpr<T>, rhs: KExpr<T>): KExpr<T> {
-    // todo: evaluate arith power
+    val lValue = lhs.toRealValue()
+    val rValue = rhs.toRealValue()
+
+    if (lValue != null && rValue != null) {
+        tryEvalArithPower(lValue, rValue)?.let { return castRealValue(it, lhs.sort) }
+    }
+
+    if (lValue == RealValue.one) {
+        return lhs
+    }
+
+    if (rValue == RealValue.one) {
+        return lhs
+    }
+
     return mkArithPowerNoSimplify(lhs, rhs)
 }
 
+private fun tryEvalArithPower(base: RealValue, power: RealValue): RealValue? = when {
+    base.isZero() && power.isZero() -> null
+    power.isZero() -> RealValue.one
+    base.isZero() -> RealValue.zero
+    base == RealValue.one -> RealValue.one
+    power == RealValue.one -> base
+    power == RealValue.minusOne -> base.inverse()
+    else -> power.smallIntValue()?.let { powerIntValue ->
+        if (powerIntValue >= 0) {
+            base.pow(powerIntValue)
+        } else {
+            base.inverse().pow(powerIntValue.absoluteValue)
+        }
+    }
+}
```

Comment:
Can't we process NaNs here?

Reply:
Arithmetic doesn't support `NaN` special values. `NaN`s are part of the Fp theory.

Reply:
Right, my bad :)